### PR TITLE
Feat/fluid icon as webcomponent

### PR DESCRIPTION
**Pull request by peter-affenzeller**
#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/fluid-elements/icon/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# fluid-icon
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `ng test fluid-icon` to execute the unit tests via
```
Comment (lukasholzer) [inline libs/fluid-elements/icon/README.md:7]:
```suggestion
Run `bazel test //libs/fluid-elements/icon:test` to execute the unit tests via
```

File: libs/fluid-elements/icon/src/lib/icon.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (!customElements.get('fluid-icon')) {
+      customElements.define('fluid-icon', FluidIcon);
+    }
+    document.body.innerHTML = '<fluid-icon name="overview"></fluid-icon>';
+    fixture = document.querySelector<FluidIcon>('fluid-icon')!;
+  });
+
+  it('should create the icon', async () => {
+    expect(fixture).not.toBe(null);
+  });
```
Comment (lukasholzer) [inline libs/fluid-elements/icon/src/lib/icon.spec.ts:32]: I guess we need more tests please add them or create follow up issue for it!

File: libs/fluid-elements/icon/src/lib/icon.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from 'lit-element';
+
+const ARROW_DOWN_ICON = html`<svg
```
Comment (lukasholzer) [inline libs/fluid-elements/icon/src/lib/icon.ts:27]: please add follow up issue for fetching the icon svg via http (simple `fetch`).

**Review by lukasholzer — CHANGES_REQUESTED** [review]

File: libs/fluid-elements/icon/src/lib/icon.ts (hunk 3 above)
Comment (peter-affenzeller) [reply libs/fluid-elements/icon/src/lib/icon.ts:27]: I created a follow-up issue to discuss how we should proceed with the icons #1635

**Review by heartdisease — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]

**Comment by lukasholzer** [comment]
@peter-affenzeller please rebase and squash your commits.

File: libs/fluid-elements/icon/src/lib/icon.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+@customElement('fluid-icon')
+export class FluidIcon extends LitElement {
+  /** Returns styles for FluidIcon component. */
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        display: inline-block;
+
+        --fluid-icon--primary-color: var(--color-primary-100);
```
Comment (peter-affenzeller) [inline libs/fluid-elements/icon/src/lib/icon.ts:57]: NIT: I'd rename the property to just `--fluid-icon--color`, since there is no secondary color 😉

**Review by heartdisease — DISMISSED** [review]

**Review by heartdisease — APPROVED** [review]

**Review by bardosmisi — APPROVED** [review]

**Comment by lukasholzer** [comment]
@peter-affenzeller can you please resolve the conflicts?

File: libs/fluid-elements/icon/README.md (hunk 1 above)
Comment (lukasholzer) [reply libs/fluid-elements/icon/README.md:7]: @heartdisease why have you resolved it? it is not incorporated yet.

Comment (heartdisease) [reply libs/fluid-elements/icon/README.md:7]: Seems like Github messed up the "Commit suggestion" operation...

**Review by heartdisease — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]

**Comment by lukasholzer** [comment]
manual cherry-picked into `8.x` and `8.0.x`

**Pull request merged**